src/group_theory/perm/fin.lean

```diff
@@ -226,6 +248,32 @@ i.cycle_range.injective (by simp)
   i.cycle_range.symm j.succ = i.succ_above j :=
 i.cycle_range.injective (by simp)
 
+lemma is_cycle_cycle_range {n : ℕ} {i : fin (n + 1)} (h0 : i ≠ 0) : is_cycle (cycle_range i) :=
```

Comment:
Would it make sense to have `i : fin n`, since if `i : fin 0` then the result is trivial?

Reply:
I think the `i ne 0` would be impossible to state.